cmake/find/Findtcl.cmake

```diff
@@ -0,0 +1,41 @@
+find_path(
```

Comment:
If we are using `pkgconfig-export-targets.cmake.in` (see below) then this module is not needed, package can be imported by `PkgConfig::tcl`, see example:
* https://docs.hunter.sh/en/latest/packages/pkg/libpcre.html

Reply:
tried it, and it's not working for this project at least without a complete rewrite of tcl.pc

Reply:
> it's not working for this project

What are the errors? May be [hunter_pkgconfig_export_target](https://github.com/ruslo/hunter/blob/master/cmake/modules/hunter_pkgconfig_export_target.cmake) need to be improved?

Reply:
Here is a tcl.pc file after running ./configure on a Mac, building __static__ lib
```
# tcl pkg-config source file

prefix=/Users/user/workspace/mobile/hunter/_Base/xxxxxxx/2dec4d1/c0825ff/Build/tcl/Install
exec_prefix=/Users/user/workspace/mobile/hunter/_Base/xxxxxxx/2dec4d1/c0825ff/Build/tcl/Install
libdir=/Users/user/workspace/mobile/hunter/_Base/xxxxxxx/2dec4d1/c0825ff/Build/tcl/Install/lib
includedir=${prefix}/include

Name: Tool Command Language
Description: Tcl is a powerful, easy-to-learn dynamic programming language, suitable for a wide range of uses.
URL: http://www.tcl.tk/
Version: 8.6.8
Requires.private: zlib >= 1.2.3
Libs: -L${libdir} -ltcl8.6 -ltclstub8.6
Libs.private:  -lz  -lpthread 
Cflags: -I${includedir}
```

1. `Requires.private: zlib >= 1.2.3` if I don't add package zlib, I get error __--   Package 'zlib', required by 'tcl', not found__ otherwise  __Target "tcl_test" links to target "PkgConfig::zlib" but the target was not found.__ But there is really no need in hunter zlib, as configure tests for system zlib and if not found can use one provided with the package. 

2. We are building __static__ lib but dependencies are not in __Libs:__ thet are in __Libs.private:__ so  they are not included in _LDFLAGS or INTERFACE_LINK_LIBRARIES

3. There is no dependency on the __libm__, that breaks testing on Linux

Reply:
> But there is really no need for hunter zlib, as configure tests for system zlib and if not found can use one provided with the package

It should be always zlib from Hunter:
* https://docs.hunter.sh/en/latest/creating-new/create/cmake-dependencies.html#conflict
* https://docs.hunter.sh/en/latest/faq/interaction-with-other-package-managers.html

> Target "tcl_test" links to target "PkgConfig::zlib" but the target was not found

I think this part is already handled quite well for autotools, take a look at `DEPENDS_ON_PACKAGES` variable, e.g.:
* https://github.com/ruslo/hunter/blob/2964970633d385c1d15c291e238eb099a3e86b16/cmake/projects/x11/hunter.cmake#L26-L40

> Dependencies are not in Libs: thet are in Libs.private

I don't know the details but I guess private libraries should not be used anywhere, if you got linker errors related to such libraries then they are not private - `tcl.pc` issue.

> There is no dependency on the libm, that breaks testing on Linux

Yes, that's a `tcl.pc` issue too.